backend/src/Designer/Controllers/Preview/V3/OldInstancesController.cs

```diff
@@ -0,0 +1,228 @@
+        return Ok();
+    }
+
+    /// <summary>
+    /// Action for getting a mocked response for the current task connected to the instance
+    /// </summary>
+    /// <returns>The processState</returns>
+    [HttpGet("{partyId}/{instanceGuid}/process")]
+    public async Task<ActionResult<AppProcessState>> Process(
+            [FromRoute] string org,
+            [FromRoute] string app,
+            [FromRoute] int partyId,
+            [FromRoute] Guid instanceGuid,
+            CancellationToken cancellationToken)
+    {
+        string developer = AuthenticationHelper.GetDeveloperUserName(httpContextAccessor.HttpContext);
+        string refererHeader = Request.Headers["Referer"];
+        Uri refererUri = new(refererHeader);
+        string layoutSetName = HttpUtility.ParseQueryString(refererUri.Query)["selectedLayoutSet"];
+        layoutSetName = string.IsNullOrEmpty(layoutSetName) ? null : layoutSetName;
+        Instance instance = await previewService.GetMockInstance(org, app, AuthenticationHelper.GetDeveloperUserName(httpContextAccessor.HttpContext), partyId, layoutSetName, CancellationToken.None);
+        List<string> tasks = await previewService.GetTasksForAllLayoutSets(org, app, developer, cancellationToken);
+        AppProcessState processState = new(instance.Process)
+        {
+            ProcessTasks = tasks != null
+                ? new List<AppProcessTaskTypeInfo>(tasks?.ConvertAll(task => new AppProcessTaskTypeInfo { ElementId = task, AltinnTaskType = "data" }))
+                : null
+        };
+
+        return Ok(processState);
+    }
+
+    /// <summary>
+    /// Action for getting a mocked response for the next task connected to the instance
+    /// </summary>
+    /// <returns>The processState object on the global mockInstance object</returns>
+    [HttpGet("{partyId}/{instanceGuid}/process/next")]
+    public async Task<ActionResult> ProcessNextAsync(
+            [FromRoute] string org,
+            [FromRoute] string app,
+            [FromRoute] int partyId,
+            [FromRoute] Guid instanceGuid,
+            CancellationToken cancellationToken
+    )
+    {
+        string refererHeader = Request.Headers["Referer"];
+        Uri refererUri = new(refererHeader);
+        string layoutSetName = HttpUtility.ParseQueryString(refererUri.Query)["selectedLayoutSet"];
+        layoutSetName = string.IsNullOrEmpty(layoutSetName) ? null : layoutSetName;
+        Instance instance = await previewService.GetMockInstance(org, app, AuthenticationHelper.GetDeveloperUserName(httpContextAccessor.HttpContext), partyId, layoutSetName, CancellationToken.None);
+        return Ok(instance.Process);
+    }
+
+    /// <summary>
+    /// Action for mocking an end to the process in order to get receipt after "send inn" is pressed
+    /// </summary>
+    /// <returns>Process object where ended is set</returns>
+    [HttpPut("{partyId}/{instanceGuid}/process/next")]
+    public async Task<ActionResult> UpdateProcessNext(
```

Comment:
```suggestion
    public ActionResult UpdateProcessNext(
```

Reply:
I think this still has to be async as there is an await operator on [line 164](https://github.com/Altinn/altinn-studio/pull/14154/files#diff-160c82f9ef0b3b995b63211b27554608a95ca621f95efb03e00541443b4dcde9R164)